### ISSUE TITLE
Properties not removed after running recipe RemoveRedundantDependencyVersions

### ISSUE THREAD

**Issue by wesboe** [openrewrite/rewrite#4350]
## What problem are you trying to solve?
The RemoveRedundantDependencyVersions recipes removes redundant version of dependencies. But when the version is a property, the property is not removed. 
**Impact**
Unused properties in your pom.xml or you override the property of a parent pom. 
For example Spring Boot has [Managed Dependency Coordinates](https://docs.spring.io/spring-boot/appendix/dependency-versions/coordinates.html) in their parent pom spring-boot-dependencies.

## What precondition(s) should be checked before applying this recipe?
We can limit the execution to Maven.

## Describe the situation before applying the recipe
```maven
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.3.2</version>
    </parent>

    <artifactId>example</artifactId>

    <properties>
        <logback.version>1.4.7</logback.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
    </dependencies>
</project>
```

## Describe the situation after applying the recipe

```java
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.3.2</version>
    </parent>

    <artifactId>example</artifactId>

    <properties>
    </properties>

    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
    </dependencies>
</project>
```

## Have you considered any alternatives or workarounds?
For the properties managed by Spring Boot you can use the recipe to remove a property tag and add all properties managed by Spring Boot.

## Are you interested in [contributing this recipe to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#adding-new-recipes)?
Yes I would like to contribute

**Comment by HelloDhero** [comment]
I think if this is deleted, it will bring many side effects

**Comment by timtebeek** [comment]
There is now a separate RemoveUnusedProperties that you can optionally run:
- https://docs.moderne.io/user-documentation/recipes/recipe-catalog/maven/removeunusedproperties/
- https://github.com/openrewrite/rewrite/blob/main/rewrite-maven/src/main/java/org/openrewrite/maven/RemoveUnusedProperties.java

Hope that helps you there!

**Comment by DidierLoiseau** [comment]
@timtebeek I don’t think that covers the same need because `RemoveUnusedProperties` only covers _unused_ properties, while this ticket was about removing _used_ properties that are already defined in the parent.

There is however [`RemoveRedundantProperties`](https://docs.openrewrite.org/recipes/maven/removeredundantproperties) that can do that as well – although it does not manage versions comparison.

Maybe the problem is more relevant when:

* you explicitly specify a version for a dependency that is managed elsewhere (parent or imported pom)
* you specify it with a property with a different name than in the parent

In that case, you would need an explicit invocation to `RemoveUnusedProperties` after `RemoveRedundantDependencyVersions`. The problem I see is that `RemoveUnusedProperties` might break cases where the property is defined in a parent pom and used in a child pom/project – possibly defined in a different repository.

(I think I had subscribed to this ticket a long time ago so I’m not too sure what exact case we were facing at the time)

**Comment by timtebeek** [comment]
Thanks for the nuance there! @wesboe any details about whether either of these satisfies the need you had there? Or whether any work like described above is still necessary?

I imagine it's common for Spring Boot projects to use properties to override versions of managed dependencies, with a bit of a challenge when to definitively remove which properties as we change parents/boms, without upsetting anyone.
